MatrixSDK/Crypto/MXCrypto.m

```diff
@@ -653,7 +661,12 @@ - (MXEventDecryptionResult *)decryptEvent2:(MXEvent *)event inTimeline:(NSString
         result = [alg decryptEvent:event inTimeline:timeline];
         if (result.error)
         {
-            MXLogError(@"[MXCrypto] decryptEvent: Error for %@: %@\nEvent: %@", event.eventId, result.error, event.JSONDictionary);
+            NSDictionary *details = @{
```

Comment:
Annoyingly objc will crash if we ever add `nil` value to a dictionary, meaning it is safer to always default to something rather than risk logs crashing the app

Reply:
Whereabouts does this crash actually happen? I'm wondering if there would be an opportune place we could map all of the `nil` values in the dictionary into `@"unknown"` before the crash takes place?

Otherwise it kind of feels like it would be an easy place for someone to introduce a crash by either not knowing this pitfall, or by not realising a value is nullable.

Reply:
It is the actual construction of dictionary, for example this will crash:
```objc
NSString *value = nil;
NSDictionary *dict = @{
  @"value": value // If I passed nil here directly the compiler would catch it
};
```

So we can't really put any kind of safety wrapper around it. The rest of the codebase has this issue as well but is seems we rarely have unexpected nil values. My justification for allowing this was that most new code we write is in swift and thus we won't be really adding new dictionary contexts in objective-c, but of course I could be wrong

Reply:
TIL that this is standard objc behaviour! Ignore me 🤐